In [113]:
import pandas as pd  
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns


In [114]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [115]:
gdp_df

Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2020  1970.560169              NaN
1        Afghanistan  2019  2065.036235              NaN
2        Afghanistan  2018  2033.804389              NaN
3        Afghanistan  2017  2058.400221              NaN
4        Afghanistan  2016  2057.067978              NaN
...              ...   ...          ...              ...
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN

[7176 rows x 4 columns]

In [116]:
gdp_df.shape

(7176, 4)

Q. 6) there are 7176 row and  4 coulmns in the data frame 

In [117]:
gdp_df.dtypes

Country or Area     object
Year                 int64
Value              float64
Value Footnotes    float64
dtype: object

country= oject aka text, year= int, value = decimal, value footnote= decimal, dype= text 

In [118]:
gdp_df=gdp_df.drop(columns=['Value Footnotes'])

In [119]:
gdp_df

Country or Area  Year        Value
0        Afghanistan  2020  1970.560169
1        Afghanistan  2019  2065.036235
2        Afghanistan  2018  2033.804389
3        Afghanistan  2017  2058.400221
4        Afghanistan  2016  2057.067978
...              ...   ...          ...
7171        Zimbabwe  1994  2603.927369
7172        Zimbabwe  1993  2420.092791
7173        Zimbabwe  1992  2437.152908
7174        Zimbabwe  1991  2733.696626
7175        Zimbabwe  1990  2652.129209

[7176 rows x 3 columns]

In [120]:
#changed the coulmn names 
gdp_df = gdp_df.rename(columns={"Country or Area": "Country", "Value": "GDP_Per_Capita"})

In [121]:
gdp_df.Country.value_counts()

Country
Least developed countries: UN classification    31
Niger                                           31
Middle East & North Africa (IDA & IBRD)         31
Middle income                                   31
Mongolia                                        31
                                                ..
Kosovo                                          13
Turks and Caicos Islands                        10
Sint Maarten (Dutch part)                       10
Somalia                                          8
Djibouti                                         8
Name: count, Length: 242, dtype: int64

In [125]:
gdp_df.Country.value_counts().to_frame()

count
Country                                            
Least developed countries: UN classification     31
Niger                                            31
Middle East & North Africa (IDA & IBRD)          31
Middle income                                    31
Mongolia                                         31
...                                             ...
Kosovo                                           13
Turks and Caicos Islands                         10
Sint Maarten (Dutch part)                        10
Somalia                                           8
Djibouti                                          8

[242 rows x 1 columns]

In [127]:
country_count=gdp_df.Country.value_counts().to_frame()

In [129]:
country_count.query('count==31')

count
Country                                            
Least developed countries: UN classification     31
Niger                                            31
Middle East & North Africa (IDA & IBRD)          31
Middle income                                    31
Mongolia                                         31
...                                             ...
IDA only                                         31
Fragile and conflict affected situations         31
Ireland                                          31
France                                           31
Italy                                            31

[205 rows x 1 columns]

205 countries have data for all years 

In [131]:
missing_years=country_count.query('count!=31')

In [133]:
missing_years.shape

(37, 1)

37 countries have mising years 